src/server/rest/v1/service/UtilsService.ts

```diff
@@ -2025,4 +2005,99 @@ export default class UtilsService {
     }
     return tag;
   }
+
+  private static buildSiteAreaTree(currentSiteArea: SiteArea, siteAreas: SiteArea[]): SiteArea {
```

Comment:
This method is also used by the smart charging integration. So I think it should not be placed in UtilsService and should not be private.
I saw that you removed the s for multiple trees from the method name. The reason for the s was that there can be multiple trees on one site. E.G. Site: Mougins --> 3 root site areas --> 3 independent Trees which are build with this method